zebra-network/src/protocol/external/tests/prop.rs

```diff
@@ -0,0 +1,80 @@
+use bytes::BytesMut;
+use proptest::{collection::vec, prelude::*};
+use tokio_util::codec::{Decoder, Encoder};
+
+use zebra_chain::serialization::{
+    SerializationError, ZcashDeserializeInto, ZcashSerialize, MAX_PROTOCOL_MESSAGE_LEN,
+};
+
+use super::super::{Codec, InventoryHash, Message};
+
+/// Maximum number of random input bytes to try to deserialize an [`InventoryHash`] from.
+///
+/// This is two bytes larger than the maximum [`InventoryHash`] size.
+const MAX_INVENTORY_HASH_BYTES: usize = 70;
+
+proptest! {
+    /// Test if [`InventoryHash`] is not changed after serializing and deserializing it.
+    #[test]
+    fn inventory_hash_roundtrip(inventory_hash in any::<InventoryHash>()) {
+        let mut bytes = Vec::new();
+
+        let serialization_result = inventory_hash.zcash_serialize(&mut bytes);
+
+        prop_assert!(serialization_result.is_ok());
```

Comment:
Future Bug Risk: Let's make sure we update `MAX_INVENTORY_HASH_BYTES` if the maximum size ever changes.

```suggestion
        prop_assert!(serialization_result.is_ok());
        prop_assert!(serialization_result.unwrap().len() <= MAX_INVENTORY_HASH_BYTES);
```

Reply:
Done in [627fbec](https://github.com/jvff/zebra/commit/627fbecc013897e10d8907a67ed1610d82ff1d45). I also made it `<` because I made the constant larger than the maximum `InventoryHash` size. This was just to be sure we covered what happened if we have an input that has trailing bytes, and that it doesn't cause any issues.